packages/grafana-ui/src/components/Text/utils.ts

```diff
@@ -30,3 +30,31 @@ export const customColor = (color: TextProps['color'], theme: GrafanaTheme2): st
       return color ? theme.colors.text[color] : undefined;
   }
 };
+
+export const customVariant = (
+  theme: GrafanaTheme2,
+  element: TextProps['element'],
+  variant?: keyof ThemeTypographyVariantTypes
+) => {
+  if (variant) {
+    return theme.typography[variant];
+  }
+  switch (element) {
+    case 'span':
+      return;
```

Comment:
Let's leave a comment here explaining why span doesnt have a default as well, something like we want the default `Text` style to just take the parent's variant.